prody/proteins/pdbfile.py

```diff
@@ -590,16 +593,34 @@ def _parsePDBLines(atomgroup, lines, split, model, chain, subset,
             resnames[acount] = resname
             chainids[acount] = chid
             if isPDB:
-                try:
-                    resnums[acount] = int(line[22:26])
-                except ValueError:
+                if dec:
+                    try:
+                        resnum = int(line[22:26])
+                    except ValueError:
+                        dec = False
+
+                    if resnums[acount-2] >= MAX_N_RES:
+                        dec = False
+                else:
                     try:
-                        resnums[acount] = int(line[22:26], 16)
+                        resnum = int(line[22:26], 16)
+                        
                     except ValueError:
                         LOGGER.warn('failed to parse residue number in line {0}. Assigning it by incrementing.'
                                     .format(i))
-                        resnums[acount] = resnums[acount-1]+1
-                icodes[acount] = line[26] 
+                        resnum = resnums[acount-1]+1
```

Comment:
We need to safeguard the case where acount is 0 here.

Reply:
I've done this now.

Reply:
Oh actually, I hadn't. Now have.